lib/parse-dir.js

```diff
@@ -1,168 +1,183 @@
+      case '.ls':
+      case '.livescript':
+      case '.js':
+        jsParser(meta, done);
+        break;
+      default:
+        return done();
+    }
+  }
+
+  processFile (file, callback) {
+    const isAsync = this.__async;
+
+    async.auto({
+      __async: (done) => {
+        return done(null, isAsync);
       },
-      function(done) {
-        // Get the files raw contents
-        if (meta.isDirectory) {
-          return done();
-        }
-        var onFileRead = function (err, fileData) {
-          var error = null;
-
-          if (err || !fileData) {
-            error = err || true;
-          }
-
-          if (!error) {
-            meta.raw = fileData.toString();
-          }
-
-          done(error);
-        };
-
-        if (that.__async) {
-          fs.readFile(meta.filepath, onFileRead);
-        } else {
-          onFileRead(null, fs.readFileSync(meta.filepath));
-        }
+      meta: (done) => {
+        return done(null, {
+          relativePath: file,
+          filepath: path.resolve(file),
+          filename: path.basename(file),
+          basename: path.basename(file, path.extname(file)),
+          extension: path.extname(file),
+          parsed: false
+        });
       },
-      function(done) {
-        // Parse the file using the correct lib
-
-        if (meta.isDirectory) {
-          return done();
+      isDirectory: ['meta', '__async', this.__determineIfDirectory],
+      fileContents: ['isDirectory', this.__getFileContents],
+      parsedContents: ['fileContents', this.__parseFileContents]
+    },
+    (err, results) => {
+      const resultObject = results.meta;
+      if (!err) {
+        resultObject.isDirectory = results.isDirectory;
+        resultObject.raw = results.fileContents;
+        if (results.parsedContents) {
+          resultObject.contents = results.parsedContents;
+          resultObject.parsed = true;
+        } else {
+          if (!resultObject.isDirectory) {
+            resultObject.contents = resultObject.raw;
+          }
         }
-
-        switch (meta.extension) {
-          case '.json':
-            jsonParser(meta, done);
-            break;
-          case '.yml':
-          case '.yaml':
-            yamlParser(meta, done);
-            break;
-          case '.coffee':
-          case '.litcoffee':
-          case '.ls':
-          case '.livescript':
-          case '.js':
-            jsParser(meta, done);
-            break;
-          default:
-            done();
-            break;
+      }
+      this.data.push(resultObject);
+      //TODO: errors aren't bubbling up
+      //need to think about this since we don't want to throw a parse error.  or do we?
+      callback();
+    });
+  }
+}
+
+// front-end to parse a directory at a time:
+class ParseDir {
+  constructor(options) {
+    this.__async = options.async !== undefined ? options.aync : true;
+    this.fileParser = new FileProcessor(options);
+  }
+
+  sync (pattern) {
```

Comment:
I think we should normalise spacing between methods too. This class gets spaces between methods but not the `FileProcessor` class. My preference is to use more new lines in general as things feel less packed for me. 

@jgallen23 thoughts on this?